Dockerfile

```diff
@@ -0,0 +1,30 @@
+FROM rust:latest as build
+
+ARG SYNAPSE_VERSION=master
+ARG SYNAPSE_GIT_SRC=https://github.com/Luminarys/synapse.git
+
+WORKDIR /usr/src
+
+RUN git clone $SYNAPSE_GIT_SRC . \
+  && [ "$SYNAPSE_VERSION" != 'master' ] && git checkout tags/$SYNAPSE_VERSION || git checkout master \
+  ;
+RUN cargo build --release --all
+
+FROM debian:latest
+
+ENV SYNAPSE_HOME=/opt/synapse
+
+RUN apt-get update \
+  && apt-get -y install libssl1.1 \
+  && apt-get clean \
+  && rm -rf /var/lib/apt/lists/* \
+  && useradd --comment "Synapse" --home ${SYNAPSE_HOME} --shell /usr/sbin/nologin --system synapse \
+  && mkdir -p ${SYNAPSE_HOME} \
+  && chown synapse:synapse ${SYNAPSE_HOME} \
+  ;
+
+COPY --from=build /usr/src/target/release/synapse /usr/src/target/release/sycli /usr/local/bin/
+
+USER synapse
+WORKDIR $SYNAPSE_HOME
+CMD ["synapse"]
```

Comment:
This should probably `FROM scratch` and copy over only the files necessary to run Synapse

Reply:
That'll only work if synapse can be statically compiled. Is that possible?

Reply:
It'll work even if you don't statically link synapse. You just copy in its dependencies, too. But yes, you can statically link synapse.

Reply:
Then that would a good approach for a completely optimised prod image.